### PR TITLE
fix: (core) added selected button with group

### PR DESCRIPTION
**Pull request by stefanoScalzo**
#### Please provide a link to the associated issue.
fixes:#1542
fixes: #1211
#### Please provide a brief summary of this pull request.
Allow the button group to have one option selected and toggle as each button gets clicked

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done

Documentation checklist:
- [x] Documentation Examples

### CODE REVIEW

**Comment by droshev** [comment]
from the changes I see, this is not just docs change

**Comment by stefanoScalzo** [comment]
> from the changes I see, this is not just docs change

Changed!

**Comment by droshev** [comment]
@stefanoScalzo Can you add some description on the button group examples - the 2 use cases we support?

File: apps/docs/src/app/core/component-docs/calendar/examples/calendar-i18n-examples/calendar--i18n-moment-example.component.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    isSelected(language: string): string {
+        return moment.locale() === language ? 'selected' : '';
+    }
+
+    setLocale(language: string): void {
+        moment.locale(language);
+        this.calendarI18n.i18nChange.next();
+    }
+
+}
```
Comment (JKMarkowski) [inline apps/docs/src/app/core/component-docs/calendar/examples/calendar-i18n-examples/calendar--i18n-moment-example.component.ts]: Do we need to add this file ?

File: apps/docs/src/app/core/component-docs/calendar/examples/calendar-i18n-examples/calendar-i18n-example.component.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        private calendarI18n: CalendarI18n,
+        private calendarI18nLabels: CalendarI18nLabels
+    ) { }
+
+    ngAfterViewInit(): void {
+        this.calendarI18n.i18nChange.next();
+        this.calendarI18nLabels.labelsChange.next();
+    }
+
+}
```
Comment (JKMarkowski) [inline apps/docs/src/app/core/component-docs/calendar/examples/calendar-i18n-examples/calendar-i18n-example.component.ts]: Do we need to add this file ?

File: apps/docs/src/app/core/component-docs/calendar/examples/calendar-i18n-examples/i18n-readonly-properties.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        if (!this.monthsShort || this.monthsShort.length === 0) {
+            this.monthsShort = this.monthsShortFallback;
+        }
+
+        if (!this.monthsFull || this.monthsFull.length === 0) {
+            this.monthsFull = this.monthsFullFallback;
+        }
+    }
+
+}
```
Comment (JKMarkowski) [inline apps/docs/src/app/core/component-docs/calendar/examples/calendar-i18n-examples/i18n-readonly-properties.ts]: Do we need to add this file ?

Comment (stefanoScalzo) [reply apps/docs/src/app/core/component-docs/calendar/examples/calendar-i18n-examples/i18n-readonly-properties.ts]: they shouldnt be there ill remove them

**Comment by JKMarkowski** [comment]
Hi @stefanoScalzo  could you try to rebase this branch to `origin/master`? In this PR I see a lot of code already existing in master.

**Pull request merged**